src/main/java/duke/Storage.java

```
@@ -0,0 +1,133 @@
+                System.out.println("File is loading...");
+            } else {
+                file.createNewFile();
+                System.out.println("File already exists!");
+            }
+        } catch (IOException e) {
+            throw new DukeException("☹ OOPS!!! File creation error");
+        }
+    }
+
+    /**
+     * Returns a list of tasks in the myDuke.txt file.
+     *
+     * @return a list of tasks.
+     * @throws DukeException due to FileNotFoundException.
+     */
+    public ArrayList<Task> displayTasks() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(file);
+            while (s.hasNextLine()) {
+                Task task = parseTasks(s.nextLine());
+                tasks.add(task);
+            }
+        } catch (FileNotFoundException e) {
+            throw new DukeException("☹ OOPS!!! File not found!");
+        }
+        return tasks;
+    }
+
+    /**
+     * Parses the task saved in the file to identify
+     * the type of task.
+     *
+     * @param task description of task in String.
+     * @return a newly created Task.
+     * @throws DukeException if the type of task is unknown.
+     */
+    public Task parseTasks(String task) throws DukeException {
+        String[] description = task.split(" / ", 4);
+        Task myTask = null;
+        try {
+            LocalDateTime dateTime;
+            String type = description[0];
+            switch (type) {
+            case "T":
+                myTask = new ToDo(description[2]);
+                break;
+
+            case "D":
+                dateTime = LocalDateTime.parse(description[3]);
+                myTask = new Deadline(description[2], dateTime);
+                break;
+
+            case "E":
+                dateTime = LocalDateTime.parse(description[3]);
+                myTask = new Event(description[2], dateTime);
+                break;
+            default:
```

Hi, maybe you should throw an exception here? Leaving it blank may result in future bugs.